src/browser/provider/cache-item.js

```diff
@@ -0,0 +1,32 @@
+/* global Symbol */
+import Promise from 'pinkie';
+import BrowserProviderPluginHost from './plugin-host';
+
+
+const initPromise = Symbol();
+
+export default class extends BrowserProviderPluginHost {
```

Comment:
`export default class BrowserProviderCacheItem extends BrowserProviderPluginHost`

BTW, not sure I understand how it works, why it extends `BrowserProviderPluginHost`? Shouldn't it just store already constructed provider?